### ISSUE TITLE
USA Drop Zone grants more money when money crate falls on different home building (?)

### ISSUE THREAD

**Issue by xezon** [xezon/GeneralsGamePatch#151]
USA Drop Zone grants more money when money crate falls on different home building (?). Reported by Mad Rage. Investigate.

**Comment by Jundiyy** [comment]
Yes this is true, it's to do with the geometry of the box, could be fixed but then it also means any dropped crates from planes or when the building has been destroyed, won't be picked up easily.

**Comment by ThePredatorBG** [comment]
The number of times crate money is collected corresponds to the number of objects (units/buildings) the crate collides with in the same frame.
So, if a crate provides $250 and collides with two structures in the same frame, the player will collect double the amount: $500.

One way to fix that, as Jundiyy mentioned, is by changing the crate geometry. A value of 0.1 will ensure the crate is collected by one object only, but the geometry of the crate becomes smaller, and the crate will be harder to collect.
A better solution (untested) might be to have the crate with 0.1 geometry initially and get its vanilla geometry back after time. This might be possible by using DynamicGeometryInfoUpdate/FirestormDynamicGeometryInfoUpdate or adding a lifetime after which a different crate with proper geometry will spawn. The lifetime must be determined by measuring the time it takes from the crate spawn to the moment the crate touches the ground in-game. This way the chance of crates being picked up by multiple objects is none, or close to none, while the crates will have a normal geometry when they collide with units.

**Comment by xezon** [comment]
After change #1390, it appears crates no longer fall off the Drop Zone, so perhaps this is no longer a practical issue?

**Comment by commy2** [comment]
Should be fixed in engine anyway.
